product_import/wizard/product_import.py

```diff
@@ -0,0 +1,209 @@
+        yesterday = today - timedelta(days=1)
+        seller_id = False
+        result = []
+        if product:
+            # Terminate previous prices
+            for s_info in product.seller_ids:
+                if s_info.name.id != seller_info["name"]:
+                    continue
+                if s_info.date_end and s_info.date_end < today:
+                    continue
+                if (
+                    s_info.min_qty == seller_info["min_qty"]
+                    and s_info.price == seller_info["price"]
+                    and s_info.currency_id.id == seller_info["currency_id"]
+                ):
+                    seller_id = s_info.id
+                else:
+                    result.append((1, s_info.id, {"date_end": yesterday}))
+        if not seller_id:
+            seller_info.setdefault("date_start", today)
+            result.append((0, 0, seller_info))
+        return result
+
+    @api.model
+    def _prepare_product(self, parsed_product, chatter_msg, seller=None):
+        try:
+            product = self._bdimport._match_product(
+                parsed_product, chatter_msg, seller=seller
+            )
+        except UserError:
+            product = None
+        uom = self._bdimport._match_uom(parsed_product["uom"], chatter_msg)
+        currency = self._bdimport._match_currency(
+            parsed_product["currency"], chatter_msg
+        )
+        weight_uom = parsed_product.get("weight_uom") or False
+        if weight_uom:
+            weight_uom = self._bdimport._match_uom(weight_uom, chatter_msg)
+
+        product_vals = {
+            "default_code": parsed_product["code"],
+            "barcode": parsed_product["barcode"],
+            "name": parsed_product["name"],
+            "description": parsed_product["description"],
+            "type": "product",
+            "uom_id": uom.id,
+            "uom_po_id": uom.id,
+            "weight": parsed_product.get("weight") or False,
+            "weight_uom_id": weight_uom and weight_uom.id or False,
+        }
+        seller_info = {
+            "name": seller and seller.id or False,
+            "product_code": parsed_product["product_code"],
+            "price": parsed_product["price"],
+            "currency_id": currency.id,
+            "min_qty": parsed_product["min_qty"],
+        }
+        product_vals["seller_ids"] = self._prepare_supplierinfo(seller_info, product)
+        if product:
+            product_vals["recordset"] = product
+
+        return product_vals
+
+    @api.model
+    def create_product(self, parsed_product, chatter_msg, seller=None, filename=None):
+        product_vals = self._prepare_product(parsed_product, chatter_msg, seller=seller)
+        product = product_vals.pop("recordset", None)
+        if product:
+            product.write(product_vals)
+            logger.info("Product %d updated", product.id)
+        else:
+            product = self.env["product.product"].create(product_vals)
```

Comment:
We should probably collect data for all prods and call create just once.
Not a blocker, you can add a TODO.